Fix Get-DbaRegisteredServerName Group issue

wsmelton opened the pull request:
<!-- Below information IS REQUIRED with every PR -->
## Type of Change
<!-- What type of change does your code introduce -->
 - [x] Bug fix (non-breaking change, fixes #1975 )
 - [ ] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Pester test is included
<!-- Below this line you can erase anything that is not applicable -->
### Purpose
<!-- What is the purpose or goal of this PR? (doesn't have to be an essay) --> 
Fixing a bug when passing in a group to the command. The dynamic part of the parameter is not working for me anymore but it should accept the value now.
### Approach
<!-- How does this change solve that purpose -->
Found a few issues where we were checking for an empty string/whitespace in an inaccurate method. Adjusted that to properly validate if the string was null, empty, or whitespaced.

As well added proper param block to the internal command so it is more clear parameters being passed to it, and corrected spelling error on parameter name (`$Bbase` to `$Base`).

### Commands to test
<!-- if these are the examples in the help just not it as such -->
`Get-DbaRegisteredServerName -SqlInstance Server -Group SomeGroup`

`Get-DbaRegisteredServerName -SqlInstance Server`
### Screenshots
<!-- pictures say a thousand words without typing any of it -->
Passing `-Group`

![image](https://user-images.githubusercontent.com/11204251/28876842-235edaec-7760-11e7-8ca9-bf6f47b5b67d.png)

No `-Group`

![image](https://user-images.githubusercontent.com/11204251/28877037-c001c13e-7760-11e7-8650-72c7720df66d.png)

### Learning
<!-- Optional -->
<!-- 
	Include:
	 - blog post that may have assisted in writing the code
	 - blog post that were initial source
	 - special or unique approach made to solve the problem
-->
Issue was being hit at line 99 with validating two variables that contained the group being passed in. Line 93 was also wrong so updated it to validate the `$Base`.

![image](https://user-images.githubusercontent.com/11204251/28878205-3301591c-7764-11e7-8053-6a93ff62c917.png)

A bit of research and chatting with Klaas. Came across [proper way to check for empty string](https://blogs.technet.microsoft.com/heyscriptingguy/2015/10/07/dude-a-string-is-a-string-in-powershell/) and updated line 93 to be `$Base -eq $null -or [string]::IsNullOrWhiteSpace($Base)`

gbargsley commented:
Testing now.  Your two tests work, but I get errors trying to get to a group within a group.

![image](https://user-images.githubusercontent.com/169602/28878820-120c2406-7766-11e7-8b1c-ccc2722fee1b.png)

wsmelton commented:
Ok... @gbargsley 

- I've added output to better inform that it didn't find anything trying to use just the sub-group name. Right now it will only support the full path being provided to that sub-group. I can't get the logic right at the moment and would have to work through it with someone that has more smarts 🔢 

- I've also updated help to include an example of looking just for servers under a sub-folder.

- Also added test to validate output of command. It will create the following structure:

![image](https://user-images.githubusercontent.com/11204251/28889751-a9e733be-778a-11e7-92d7-6871baf0fca8.png)

potatoqualitee commented:
Thanks yall 👍 !! Love seeing all these PRs. Will resolve all tomorrow.
